cloud/scope/machine.go

```diff
@@ -127,37 +155,72 @@ func (m *MachineScope) CreateMachine() (*vpcv1.Instance, error) {
 		return nil, err
 	}
 
-	imageID, err := fetchImageID(m.IBMVPCMachine.Spec.Image, m)
-	if err != nil {
-		record.Warnf(m.IBMVPCMachine, "FailedRetriveImage", "Failed image retrival - %v", err)
-		return nil, fmt.Errorf("error while fetching image ID: %v", err)
+	options := &vpcv1.CreateInstanceOptions{}
+	// Build common field resources, as unique InstancePrototype's are defined based on machine source.
+	profile := &vpcv1.InstanceProfileIdentity{
+		Name: &m.IBMVPCMachine.Spec.Profile,
+	}
+
+	subnetIdentity := &vpcv1.SubnetIdentity{}
+	// Find subnet ID from Cluster Network Status, if possible.
+	// If Network Status is available, attempt to retrieve subnet ID from there.
+	if m.IBMVPCCluster.Status.Network != nil {
+		if subnet, ok := m.IBMVPCCluster.Status.Network.ControlPlaneSubnets[m.IBMVPCMachine.Spec.PrimaryNetworkInterface.Subnet]; ok {
+			subnetIdentity.ID = ptr.To(subnet.ID)
+		} else if subnet, ok := m.IBMVPCCluster.Status.Network.WorkerSubnets[m.IBMVPCMachine.Spec.PrimaryNetworkInterface.Subnet]; ok {
+			subnetIdentity.ID = ptr.To(subnet.ID)
+		}
+	}
+	// If the ID hasn't been set yet, rely on Machine Spec for lookup, and finally falling back to previous logic of using the subnet value directly as an ID.
+	if subnetIdentity.ID == nil {
+		// For Machines not reliant directly on Cluster managed subnets, lookup subnet ID by name.
+		subnetDetails, err := m.IBMVPCClient.GetVPCSubnetByName(m.IBMVPCMachine.Spec.PrimaryNetworkInterface.Subnet)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving subnet ID for machine %s: %w", m.IBMVPCMachine.Name, err)
+		} else if subnetDetails != nil {
+			subnetIdentity.ID = subnetDetails.ID
+		} else {
+			subnetIdentity.ID = &m.IBMVPCMachine.Spec.PrimaryNetworkInterface.Subnet
+		}
+	}
+	primaryNetworkInterface := &vpcv1.NetworkInterfacePrototype{
```

Comment:
If there is time, I may try to update this PR with the logic to perform SecurityGroup lookups (from name to ID) to complete the network interface configuration.

Reply:
Added, I will attempt to do some basic validation testing and see if I can also add one or more unit tests as well.